src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -458,6 +483,20 @@ private async Task<IReadOnlyList<RestoreSummary>> PerformNuGetV2RestoreAsync(Pac
             return configToProjectPath;
         }
 
+        private static Dictionary<string, RestoreAuditProperties> GetRestoreAuditProperties(PackageRestoreInputs packageRestoreInputs)
+        {
+            Dictionary<string, RestoreAuditProperties> restoreAuditProperties = new(PathUtility.GetStringComparerBasedOnOS());
+            foreach (PackageSpec project in packageRestoreInputs.ProjectReferenceLookup.Projects)
```

Comment:
Unless I'm using VSCode wrong, `.Projects` has type `IReadOnlyList<T>`, and therefore `foreach` will allocation.  I
```suggestion
            foreach (PackageSpec project in packageRestoreInputs.ProjectReferenceLookup.Projects.NoAllocateEnumerate())
```